code/modules/mining/mine_items.dm

```diff
@@ -1130,17 +1130,24 @@ var/list/total_extraction_beacons = list()
 	force = 15
 	throwforce = 5
 	origin_tech = list(TECH_BLUESPACE = 4, TECH_ENGINEERING = 3)
+	var/limit = 50
```

Comment:
This doesn't need to be an object variable, since it is not used outside of one function. Further, your bug of it only working once is caused by you never setting this limit back to 50.

Make it a function variable and your problem is solved.